### PR TITLE
Don't fail on skipped file/dir dependencies

### PR DESCRIPTION
**Pull request by Pwuts**
# Pull Request Check List

Resolves: #8548

- [x] Added **tests** for changed code.
- [ ] Updated **documentation** for changed code.

As suggested by @dimbleby: https://github.com/python-poetry/poetry/issues/8548#issuecomment-1767062264

### CODE REVIEW

**Comment by dimbleby** [comment]
> (plus corresponding unit test)

was the full suggestion!

**Comment by Wladastic** [comment]
I'd update this branch but I have no access currently^^

As there are no poetry tests we implemented, I guess this can be approved, but documentation could be added that dependencies can be skipped.

**Comment by dimbleby** [comment]
> As there are no poetry tests we implemented, I guess this can be approved

I don't follow what you're saying: but there are testcases that could and should be written to go with this code change.

`test_install_path_dependency_does_not_exist()` is probably a good starting point.

**Comment by Wladastic** [comment]
Given that tests/integration/test_setup.py could be augmented as well then, given we have no way to know if setup works on individual systems like Ubuntu, Debian, Windows 10, Windows 11, MacOS, etc.

**Comment by dimbleby** [comment]
I still can't follow your point: but poetry's CI pipelines run across a fairly wide range platforms.

Are you trying to make a case that there is something special about this code that should make it either untestable, or not worth testing?  What is it that you think is special about this code?

**Comment by Pwuts** [comment]
I'm not familiar with this codebase so writing tests is a challenge (with and without AI assistance). Standby while I try to figure this out.

**Comment by dimbleby** [comment]
I think you current draft is not the right direction.  I'd try to copy `test_install_path_dependency_does_not_exist()` in `test_install.py` (nb not `test_installer.py`), that looks really quite close to being what you want already.

the bit that's maybe confusing about that test - at least in my opinion - is that it non-obviously refers to example projects eg `missing_directory_dependency` is an actual project at `tests/fixtures/missing_directory_dependency`.

hopefully not _too_ hard to set up another fixture project that looks like the #8548 example, and write a test showing that it can be installed when an irrelevant extra is missing, but not when a relevant extra is missing

**Comment by Pwuts** [comment]
TODO:
- [ ] Add poetry.lock file

**Comment by Pwuts** [comment]
There we go. Thanks for the guidance @dimbleby! :)

File: tests/console/commands/test_install.py [hunk 1]
```diff
@@ -470,6 +470,26 @@ def test_install_path_dependency_does_not_exist(
             tester.execute(options)
 
 
+@pytest.mark.parametrize("options", ["", "--extras notinstallable"])
+@pytest.mark.parametrize("project", ["missing_extra_directory_dependency"])
```
[the remark this answers is not in the document]
Comment (Pwuts) [reply tests/console/commands/test_install.py]: fixed

**Comment by Pwuts** [comment]
Good to go now? :)

**Comment by dimbleby** [comment]
looks good to me, but you need someone with the power to press the merge button

**Comment by Wladastic** [comment]
Please update the branch first, looks good.

**Review by Wladastic — APPROVED** [review]

**Comment by Pwuts** [comment]
> looks good to me, but you need someone with the power to press the merge button

Oh I thought that was you :')

**Review by radoering — APPROVED** [review]

**Comment by Pwuts** [comment]
Awesome! Can I ask when we may expect a release containing this fix?

**Comment by radoering** [comment]
No plans so far. I wouldn't expect one this year.

**Pull request merged**